File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  const len = numbers.length;
  let [slow, fast] = [0, len - 1];
  while (slow < fast) {
      const curr = numbers[slow] + numbers[fast];
      if (curr === target) return [slow + 1, fast + 1];
      else if (curr > target) fast -= 1;
      else slow += 1;
  }
  return [-1, -1];
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        int l = 0;
        int r = n - 1;
        
        while (l < r) {
            int sum = nums[l] + nums[r];
            
            if (sum == target) {
                return {l + 1, r + 1};
            }
            if (sum < target) {
                l++;
            } else {
                r--;
            }
        }
        
        return {-1, -1};
    }
};